gorfc/gorfc.go

```diff
@@ -1140,8 +1141,41 @@ func (conn *Connection) GetFunctionDescription(goFuncName string) (goFuncDesc Fu
 	return wrapFunctionDescription(funcDesc)
 }
 
-// Call calls the given function with the given parameters and wraps the results returned.
-func (conn *Connection) Call(goFuncName string, params interface{}) (result map[string]interface{}, err error) {
+func setupParameter(params interface{}, funcDesc C.RFC_FUNCTION_DESC_HANDLE, funcCont C.RFC_FUNCTION_HANDLE) error {
+	paramsValue := reflect.ValueOf(params)
+	if paramsValue.Type().Kind() == reflect.Map {
```

Comment:
Suggestion idea for possible optimization and a bit more clean code can be replaced by switch case, like:

```
switch p := params.(type) {
case map[string]interface{}:
     // Related code ...
case struct:
     // Related code ...
default:
     // Related code ...
}
```

Reply:
I extracted this code to a method for readability, but I agree.